### PR TITLE
Add: benchmark for ping1d and ping360 on CI

### PR DESCRIPTION
**Pull request by RaulTrombin**
helps

### CODE REVIEW

**Comment by RaulTrombin** [comment]
@patrickelectric @joaoantoniocardoso Should we add this in the near future? Benchmarking in navigator-rs helped improve sensor readings and reliability.

File: benches/bench_hil.rs [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+        .build()
+        .unwrap()
+}
+
+const SAMPLE_PERIOD_TICK_DURATION: f64 = 25e-9;
+const FIRMWARE_MAX_TRANSMIT_DURATION: u16 = 1000;
+const FIRMWARE_MIN_TRANSMIT_DURATION: u16 = 1;
+
+fn calculate_sample_period(desired_range: f64, number_of_samples: u16, speed_of_sound: f64) -> u16 {
+    const SAMPLE_PERIOD_TICK_DURATION: f64 = 25e-9;
```
Comment (ES-Alexander) [inline benches/bench_hil.rs]:
```suggestion
```

File: benches/bench_hil.rs [hunk 2]
```diff
@@ -0,0 +1,583 @@
...
+    tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .thread_name("criterion-tokio-rt")
+        .build()
+        .unwrap()
+}
+
+const SAMPLE_PERIOD_TICK_DURATION: f64 = 25e-9;
+const FIRMWARE_MAX_TRANSMIT_DURATION: u16 = 1000;
+const FIRMWARE_MIN_TRANSMIT_DURATION: u16 = 1;
```
Comment (ES-Alexander) [inline benches/bench_hil.rs]: These seem like they should have unit information (in the variable names, or in explanatory comments), or be imported from the protocol specification or something (assuming they're available there, which I suspect they're not).

File: benches/bench_hil.rs [hunk 3]
```diff
@@ -0,0 +1,583 @@
...
+    auto_duration = auto_duration.max(min_sample_based).round();
+
+    // Clamp between FIRMWARE_MIN_TRANSMIT_DURATION and max based on sample period
+    let max_duration = calculate_transmit_duration_max(sample_period);
+    auto_duration = auto_duration
+        .max(FIRMWARE_MIN_TRANSMIT_DURATION as f64)
+        .min(max_duration as f64);
+
+    auto_duration.round() as u16
+}
```
Comment (ES-Alexander) [inline benches/bench_hil.rs]: Code with no [explanation](https://github.com/bluerobotics/ping-viewer/blob/1ed45085f3949e83b2376d7aa2d0364bf8905383/src/sensor/ping360.h#L463-L470) is hard to follow, evaluate, and maintain.
